tests/HttpHandlerTest.php

```diff
@@ -32,6 +36,18 @@ public function testWhenRouteIsNotFoundThenThrowException(): void
         $this->assertEquals('Not Found', $response->getReasonPhrase());
     }
 
+    public function testWhenRouteExistsForADifferentMethodThenReturnNotAllowed(): void
+    {
+        $dispatcher = (new RouteDispatcherFactory(self::METHOD_NOT_ALLOWED_CONFIG))->create();
+        $handler = new HttpHandler($dispatcher, $this->factory);
+        $request = $this->createRequest('POST', '/');
+
+        $response = $handler->handle($request);
+
+        $this->assertEquals(405, $response->getStatusCode());
+        $this->assertEquals('Method Not Allowed', $response->getReasonPhrase());
```

Comment:
Later we could use our package for asserting PSR Responses: https://github.com/bauhausphp/assert-psr-response :)